### PR TITLE
reload a scrape pool when detected a config change

### LINKED ISSUE #2756: Prom2: SIGHUP for config reload causes artifacts in TSDB storage
**What did you do?**
Run P2 from dev-2.0 branch side by side our P1 stack with the same config reloading and scraped targets.

**What did you expect to see?**
A stable retention similarly to P1.

**What did you see instead? Under which circumstances?**

A periodic, every 1h weird whole in our data, which massively crops up in rates.

image

This coincides with our config syncing job that sends a SIGHUP to prometheus at least every hour. The data loss corelates *exactly* with the SIHGUP log statement and with the prometheus_config_last_reload_success_timestamp

image

@fabxc tracked down the issue

### PR DESCRIPTION
**Pull request by krasi-georgiev**
the scrape pool needs to be restarted for any change in the config file (scrape interval , bearer token etc.) not just changes in the targets

closes #2756, #2830

### CODE REVIEW

File: retrieval/manager.go [hunk 1]
```diff
@@ -54,8 +55,6 @@ type ScrapeManager struct {
 
 // Run starts background processing to handle target updates and reload the scraping loops.
 func (m *ScrapeManager) Run(tsets <-chan map[string][]*config.TargetGroup) error {
-	level.Info(m.logger).Log("msg", "Starting scrape manager...")
```
Comment (Gouthamve) [inline retrieval/manager.go]: Any reason this is removed?

**Comment by Gouthamve** [comment]
LGTM

File: retrieval/manager.go (hunk 1 above)
Comment (krasi-georgiev) [reply retrieval/manager.go]: I didn't think it gives any useful info and no other manager/component has startup logs.
The TSDB has start up logs , but that one is useful as it also has another log `TSDB started`  when the initialisation has completed.

**Comment by Gouthamve** [comment]
The `scrapeConfig` change happens at update time of target sets and for a super stable targetset, that could take forever. Can we make sure the new `scrapeConfig` is applied immediately?

**Pull request closed without merging**